src/app/components/Calendar.tsx

```diff
@@ -1,22 +1,66 @@
+import React, {useState, useEffect} from "react"
 import FullCalendar from "@fullcalendar/react";
 import interactionPlugin from "@fullcalendar/interaction";
 import timeGridPlugin from "@fullcalendar/timegrid";
 import dayGridPlugin from '@fullcalendar/daygrid';
+import type { IEvent } from "@database/eventSchema";
 import Link from 'next/link';
 import { useRef } from "react";
 import { EventInstance } from "@fullcalendar/common";
 
-const Calendar = () => {
+//Interface to define full calendar event format
+interface FullCalendarEvent {
+  id: string;
+  title: string;
+  start: Date;
+}
+
+
+const Calendar = ({admin = false}) => {
+  const [events, setEvents] = useState<IEvent[]>([]);
+  const [fullCalendarEvents, setFullCalendarEvents] = useState<FullCalendarEvent[]>([]);
+  const [selectedEventId, setSelectedEventId] = useState('');
```

Comment:
I have made minor changes like rearranging the state declarations (state declarations should always stay at the top of the component). Other changes include removing modifications that does not relate to this issue and reverting package-lock change.